### PR TITLE
Fix Circadian Half-Cycler bug

### LINKED ISSUE #38: Circadian Half-Cycler from Royalty DLC prevents pawns from using MedPods
Steam Workshop user **Nick** reported that the Circadian Half-Cycler from the Royalty DLC causes pawns trying to use the MedPod to get stuck in a loop of getting in and out of the bed, without being healed.

The issue only appears from v1.1.2 onwards, so it is likely that the "gain rest" feature introduced from that update onwards is conflicting with the Half-Cycler.

### PR DESCRIPTION
**Pull request by jinh86**
Simple check that fixes the circadian half-cycler. Tested and playing with it now.

### CODE REVIEW

File: Source/Building_BedMedPod.cs [hunk 1]
```diff
@@ -347,11 +347,11 @@ private void DiagnosePatient(Pawn patientPawn)
 
             foreach (Hediff currentArtificialPartHediff in artificialPartHediffs)
             {
-                childPartsToSkip.AddRange(GetBodyPartDescendants(currentArtificialPartHediff.part));
+                childPartsToSkip.AddRange(GetBodyPartDescendants(currentArtificialPartHediff.Part));
```
Comment (sumghai) [inline Source/Building_BedMedPod.cs:350]: Ah, I see now why you changed `part` to `Part` - the latter is public and has null handling.

File: Source/Building_BedMedPod.cs [hunk 2]
```diff
@@ -604,7 +604,7 @@ public override void Tick()
             {
                 DiagnosingTicks--;
                 powerComp.PowerOutput = -DiagnosingPowerConsumption;
-                PatientPawn.needs.food.CurLevelPercentage = 1f;
+                if (PatientPawn != null) PatientPawn.needs.food.CurLevelPercentage = 1f;
```
Comment (sumghai) [inline Source/Building_BedMedPod.cs:607]: Good catch on the nullcheck for `PatientPawn`.

I'll add the braces myself after merging this PR, as omitting braces for one-line if statements is generally not good practice (detracts from code readability, and may cause problems if someone wants to stick additional code inside the `if{}` block later)

File: Source/MedPodMod.cs [hunk 3]
```diff
@@ -96,7 +96,7 @@ static class Toils_Bed_FailOnBedNoLongerUsable_CustomFailConditionForMedPods
             static MethodBase TargetMethod()
             {
                 predicateClass = typeof(Toils_Bed).GetNestedTypes(AccessTools.all)
-               .FirstOrDefault(t => t.FullName.Contains("c__DisplayClass3_0"));
+               .FirstOrDefault(t => t.FullName != null && t.FullName.Contains("c__DisplayClass3_0"));
```
Comment (sumghai) [inline Source/MedPodMod.cs:99]: Is this nullcheck necessary?

Most modders that I've seen targeting predicate methods seem to be happy with just going straight to `Contains()` without the additional nullcheck.

**Review by jinh86 — COMMENTED** [review]
I agree that's safe to expect it not to be null, but IDE was complaining so I got rid of it mostly for that :P
(talking about the Containts() on fullname)

**Review by sumghai — APPROVED** [review]
Approved for #38 after testing on my end - I'll make the minor changes later in a follow-up commit of my own.

Many thanks, @jinh86!

**Pull request merged**